demo/text_matching/predict.py

```diff
@@ -0,0 +1,125 @@
+    if args.is_pair_wise:
+        num_data = 3
+    else:
+        num_data = 2
+    inputs, outputs, program = module.context(
+        trainable=True, max_seq_len=args.max_seq_len, num_data=num_data)
+
+    # Tokenizer tokenizes the text data and encodes the data as model needed.
+    # If you use transformer modules (ernie, bert, roberta and so on), tokenizer should be hub.BertTokenizer.
+    # else tokenizer should be hub.CustomTokenizer.
+    tokenizer = hub.BertTokenizer(
+        vocab_file=module.get_vocab_path(), tokenize_chinese_chars=True)
+
+    # Load dataset
+    if args.is_pair_wise:
+        dataset = hub.dataset.DuEL(
+            tokenizer=tokenizer, max_seq_len=args.max_seq_len)
+    else:
+        dataset = hub.dataset.LCQMC(
+            tokenizer=tokenizer, max_seq_len=args.max_seq_len)
+
+    # Construct transfer learning network
+    # Use token-level output.
+    query = outputs["sequence_output"]
+    left = outputs['sequence_output_2']
+    right = outputs['sequence_output_3']
+
+    # Select fine-tune strategy
+    strategy = hub.DefaultStrategy(optimizer_name="sgd")
+
+    # Setup RunConfig for PaddleHub Fine-tune API
+    config = hub.RunConfig(
+        use_data_parallel=False,
+        use_cuda=args.use_gpu,
+        batch_size=args.batch_size,
+        checkpoint_dir=args.checkpoint_dir,
+        strategy=strategy)
+
+    # Define a text matching task by PaddleHub's API
+    # network choice: bow, cnn, gru, lstm (PaddleHub pre-defined network)
+    matching_task = hub.TextMatchingTask(
+        dataset=dataset,
+        query_feature=query,
+        left_feature=left,
+        tokenizer=tokenizer,
+        network=args.network,
+        is_pair_wise=args.is_pair_wise,
+        right_feature=right if args.is_pair_wise else None,
+        config=config)
+
+    # Data to be predicted
+    if args.is_pair_wise:
+        # pair_wise prediction data sample
+        text_pairs = [
+            [
+                "思追原来是个超级妹控，不愿妹妹嫁人，然而妹妹却喜欢一博老师;老师",  # text_1
+                "摘要:林妙可演唱的《老师》，作为中央电视台教师节晚会的开场曲。;谱曲:彭野新儿歌;歌曲时长:3分53秒;",  # text_2
+                "摘要:林妙可演唱的《老师》，作为中央电视台教师节晚会的开场曲。;谱曲:彭野新儿歌;歌曲时长:3分53秒;"  # text_3 same as text_2
+            ],
+            [
+                "儿子祝融被杀害，西天王大发雷霆，立即下令捉拿天庭三公主;儿子",  # text_1
+                "摘要:《儿子》是曹国昌1983年创作的木雕，收藏于中国美术馆。;材质：:木雕;作者：:曹国昌;中文名:儿子;创作年代：:1983年;义项描述:曹国昌木雕;标签:文化;",  # text_2
+                "摘要:《儿子》是曹国昌1983年创作的木雕，收藏于中国美术馆。;材质：:木雕;作者：:曹国昌;中文名:儿子;创作年代：:1983年;义项描述:曹国昌木雕;标签:文化;",  # text_3 same as text_2
+            ]
+        ]
+    else:
+        # point_wise prediction data sample
+        text_pairs = [
+            [
+                "请问不是您的账户吗？",  # text_1
+                "您好，请问您使用的邮箱类型是？"  # text_2
+            ],
+            [
+                "推荐个手机游戏",  # text_1
+                "手机游戏推荐"  # text_2
+            ]
+        ]
+
+    # Predict by PaddleHub's API
+    results = matching_task.predict(
```

Comment:
If you separate the matching task, you can copy the left feature inside your API. User can avoid write the left title which need to predict twice. That's not natural and confusing.

Reply:
Done